### ISSUE TITLE
Blank Screen Before resolving resolves

### ISSUE THREAD

**Issue by dcarabott** [ui-router/angular#79]
We're having issues when loading `bootstrap: [UIView]`. We have put a loader into the `ui-view` on `index.html` so that when the app loads the content of the `ui-view` is replaced by the app. The problem is that when having resolves on the landing state, the content of the `ui-view` is being removed before resolving the resolves. So there's quite a noticeable blank screen between the  spinner and the app. We have tried to play with resolve policy but the experience is not optimal. It would be best to have the loader visible until all resolves have resolved.

This issue is only noticeable on ui-router 1.0 as it is working fine on the previous version. 

Here's a POC demonstrating this issue: 
https://plnkr.co/edit/riXgCiIbowg2A0AIDbYz?p=preview

**index.html**
```
  <body>
    <ui-view>
      Loading...
    </ui-view>
.
.
```

**Home State (landing state)**
```
export const helloState = { name: 'hello', url: '/hello',  component: Hello, 
resolve:[
  { 
      token: 'test',
      resolveFn: () => {
        return new Promise((resolve)=>{
          setTimeout(()=>{
            resolve();
          },4000);
        })
      }
    }
  ] }; 
```

Sequence of events

1. Page loads displaying 'Loading...'
2. 'Loading...' disappears 
3. Blank screen for 4 seconds (resolving resolves)
4. Home Loads

**Comment by christopherthielen** [comment]
This [pr implements default content rendering](https://github.com/angular-ui/ui-router/pull/3082) and seems like it should handle this case.  

@aitboudad do you have any ideas why default content doesn't work on the bootstrapped ui-view, but does on the nested ui-view (inside people state)?  https://plnkr.co/edit/Jc04N43HmFzU9v2HMSMh?p=preview

**Comment by aitboudad** [comment]
I tried iwith a simple example without `ui-router` and I get the same behavior http://plnkr.co/edit/vDlXazYbvXBL1b7ksjQA?p=preview

**Comment by aitboudad** [comment]
see https://github.com/angular/angular/issues/4946

**Comment by christopherthielen** [comment]
@aitboudad thanks

**Comment by dcarabott** [comment]
Is there a way to workaround this issue for now?

**Comment by christopherthielen** [comment]
This was fixed in [`@uirouter/angular` version 1.0.0-beta.8](https://github.com/ui-router/angular/releases/tag/1.0.0-beta.8) via the `deferInitialRender` toggle.

See this example: https://stackblitz.com/edit/ui-router-angular-3yunap?file=app/app.module.ts
